### PR TITLE
Network Difficulty Graph

### PR DESCRIPTION
**Pull request by schmentle**
### Screenshot 
https://share.getcloudapp.com/YEupGDgB

### CODE REVIEW

**Review by Krakaw — CHANGES_REQUESTED** [review]
I can't find the X-Axis? Also please add a screenshot to the PR.

**Review by Krakaw — CHANGES_REQUESTED** [review]
It's close, it needs the shaded area underneath the line graph and use the block heights for the X-Axis.

Let's also decouple these graphs from the data, move the api call to `App.tsx` and pass the data in as a prop.

**Comment by schmentle** [comment]
Hi @Krakaw please see recording with tooltip and shaded area on the x axis https://share.getcloudapp.com/v1ueK7Yq

File: src/App.tsx [hunk 1]
```diff
@@ -3,15 +3,19 @@ import './App.css';
...
     const [latestBlocks, setLatestBlocks] = useState([]);
+    const [estimatedHashRate, setTotalDifficulty] = useState(([] as unknown) as any);
     useEffect(() => {
         try {
             fetchBlocksData(100).then((blockData) => {
                 setLatestBlocks(blockData.blocks as any);
             });
+            fetchNetworkDifficulty().then((data) => {
+                setTotalDifficulty(data);
+            });
```
Comment (Krakaw) [inline src/App.tsx]: You can move this into its own `useEffect` block

File: src/api.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+const apiURL: string = process.env.REACT_APP_EXPLORER_API_URL || "";
```
Comment (Krakaw) [inline src/api.ts]: Is this a duplicate from an old merge? Please merge development back into your branch.

**Review by Krakaw — CHANGES_REQUESTED** [review]
The component graphs seem to be very tightly coupled to their respective data. If this is the case they should be split out into something like `EstimatedHashGraph` instead of `PolygonGraph`. The best scenario would be extending the PolygonGraph with custom data so we could re-use it in the future.

**Comment by schmentle** [comment]
> The component graphs seem to be very tightly coupled to their respective data. If this is the case they should be split out into something like `EstimatedHashGraph` instead of `PolygonGraph`. The best scenario would be extending the PolygonGraph with custom data so we could re-use it in the future.

Thanks @Krakaw, should I also do this for the Bar graph chart?

**Comment by Krakaw** [comment]
> > The component graphs seem to be very tightly coupled to their respective data. If this is the case they should be split out into something like `EstimatedHashGraph` instead of `PolygonGraph`. The best scenario would be extending the PolygonGraph with custom data so we could re-use it in the future.
> 
> Thanks @Krakaw, should I also do this for the Bar graph chart?

Yup.

**Review by Krakaw — APPROVED** [review]

**Pull request merged**